src/app/models/AbstractModel.ts

```diff
@@ -106,12 +122,59 @@ export abstract class AbstractModelWithoutId<Model = Record<string, any>> {
    * @param meta Metadata
    */
   public addMetadata(meta: Meta): void {
-    this[AbstractModel.metaKey] = meta;
+    this[AbstractModel.keys.meta] = meta;
   }
 
   /** Get hidden model metadata */
   public getMetadata(): Meta {
-    return this[AbstractModel.metaKey];
+    return this[AbstractModel.keys.meta];
+  }
+
+  /**
+   * Converts the model and a list of keys, into an object containing each key
+   * value pair
+   *
+   * @param keys List of attributes to retrieve
+   */
+  private toObject(keys: string[]): Partial<this> {
+    const output: Partial<Writeable<this>> = {};
+    keys.forEach((attribute: keyof AbstractModel) => {
+      const value = this[attribute];
+      if (value instanceof Set) {
+        output[attribute] = Array.from(value);
+      } else if (value instanceof DateTime) {
+        output[attribute] = value.toISO();
+      } else if (value instanceof Duration) {
+        output[attribute] = value.as("seconds");
+      } else {
+        output[attribute] = this[attribute] as any;
+      }
+    });
+    return output;
+  }
+
+  /**
+   * Retrieves a list of all attributes associated with the model which are
+   * not the injector
+   */
+  private getModelAttributes(opts?: {
+    create?: boolean;
+    update?: boolean;
+    formData?: boolean;
```

Comment:
See previous comment, but the `formData` is really not expressing the intent well. You want to extract image related fields only, so you can send those via form data.

```suggestion
    images?: boolean;
```